torch/_dynamo/_trace_wrapped_higher_order_op.py

```diff
@@ -0,0 +1,111 @@
+import torch.utils._pytree as pytree
+from torch._C import DispatchKey
+from torch._higher_order_ops.utils import autograd_not_implemented
+
+from torch._ops import HigherOrderOperator
+from torch._subclasses import FakeTensorMode
+
+from torch.fx.experimental.proxy_tensor import ProxyTorchDispatchMode, track_tensor_tree
+from torch.utils._python_dispatch import _get_current_dispatch_mode
+
+
+__all__ = ["trace_wrapped"]
+
+
+# Trace wrapped is a higher order op meant for both invoking a bound function,
+# and for registering it as a call_function.
+# This allows us to re-enter dynamo during compiled autograd to trace (or graph break)
+# the functions as needed. While there is nothing backward specific about this op, the way it is written means
+# we can support functions in backward with complex python. It can be thought of as an allow_in_graph
+# for our aten graph. If we were to not do this, the functions would get inlined into their composing aten ops,
+# and we would lose the python state mutation.
+def trace_wrapped(*args, fn):
```

Comment:
Is there any reason to take this variadically, you only support one argument 🤔

Reply:
I went back and forth. This feels better in case we want to use it in autograd.Function, where we take multiple args.